File: frontend/src/app/model/Prediction.ts
export interface Prediction {
  userId: number;
  year: number;
  month: number;
  plantCount: number;
  fertilizerType: string;
  pruning: string;
  soilPH: number;
  avgRainfall: number;
  averageTemperature: number;
  avgHumidityPercent: number;
}

export interface GetAllPrediction {
  id: number;
  userId: number;
  year: number;
  month: number;
  plantCount: number;
  fertilizerType: string;
  pruning: string;
  soilPH: number;
  avgRainfall: number;
  averageTemperature: number;
  avgHumidityPercent: number;
  createdAt: Date;
  predictionResult: number;
  fullName: string;
  estate: string
}

File: backend/TeaLeavesProductionForecastWebAPI/Model/Prediction.cs
namespace TeaLeavesProductionForecastWebAPI.Model
{
    public class Prediction
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int Year { get; set; }
        public int Month { get; set; }
        public int  PlantCount { get; set; }
        public string FertilizerType { get; set; }
        public string Pruning { get; set; }
        public decimal SoilPH { get; set; }
        public decimal AvgRainfall { get; set; }
        public decimal  AverageTemperature { get; set; }
        public decimal AvgHumidityPercent { get; set; }
        public DateTime? CreatedAt { get; set; } = DateTime.Now;
        public decimal PredictionResult { get; set; }
    }
}
